### PR TITLE
grammar: Add constructors

### PR DESCRIPTION
**Pull request by taylorhansen**
### I have:
 - [x] Ran a test (`npm run dev && npm test`)
 - [ ] Added tests for my changes
 - [x] Documented my code thoroughly.

### CODE REVIEW

File: src/vsl/parser/nodes/constructor.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        this.optional = optional;
+        /** @type {FunctionArgument[]} */
+        this.params = params;
+        /** @type {CodeBlock} */
+        this.statments = statements;
+    }
+
+    /** @override */
+    get children() {
+        return ['access', 'optional', 'params', 'statements'];
```
Comment (vihanb) [inline src/vsl/parser/nodes/constructor.js]: children only should return child `Node` object so `access` and `optional` shouldn't be here.

**Review by vihanb — CHANGES_REQUESTED** [review]
Looks good, just the one thing with `get children()`

**Comment by vihanb** [comment]
Random Thought: do we want to reserve `self` and `super`. That way we could prevent stupid thing like `func self(){}` from hapenning.

**Comment by taylorhansen** [comment]
@vihanb imo I would leave that to something that isn't the parser.

**Review by vihanb — APPROVED** [review]

**Pull request merged**